### PR TITLE
CR002/CR046/CR072 - new Train formation, Occupancy and StopAssignment model

### PR DESCRIPTION
**Pull request by haeckerbaer**
CR for the addition of a formation / train composition model to PT and ET functional service. The new model is based on the NeTEx TRAIN as well as the VDV454 train formation model.
See short presentation with notes from the meeting in Berne and a summary/illustration of the changes as well as a rough mapping between SIRI and VDV454: 
[Pitch_CR002_CR046.pptx](https://3.basecamp.com/3256016/buckets/9672657/uploads/2464323377)
A simple ET journey example was mapped from a VDV454 AUS message containing actual formation data provided in Switzerland.

### CODE REVIEW

**Comment by haeckerbaer** [comment]
@hennoit 
Thank you for the detailed list! I corrected everything in the latest commit.
However, I couldn't find the first point (looked OK to me)

File: xsd/siri_utility/siri_types-v2.0.xsd [hunk 1]
```diff
@@ -153,4 +153,13 @@ A string containing a phrase in a natural language name that requires at least o
   </xsd:annotation>
   <xsd:restriction base="xsd:nonNegativeInteger"/>
  </xsd:simpleType>
+ <xsd:simpleType name="PercentageType">
```
Comment (Aurige) [inline xsd/siri_utility/siri_types-v2.0.xsd]: Just a note: My CR71 New Mode extension for FM also uses Percentage.... we need to align the types we use at some point

**Comment by mentzrh** [comment]
My colleague Werner Kohl did a review on this CR. He focused on the differences between VDV454 and SIRI (from slide 27). Here are his findings (I tried to translate them in English):

Slide 27:

- TrainElement: The designation is somewhat restrictive because meanwhile there are buses with trailers, i.e. consist of two vehicles. The designation as a train is also misleading for tram formations.
The term VEHICLE instead of TRAIN would probably be closer to Transmodel. This finding applies to all other element names that begin with "Train", ie "TrainElementCode", "TrainElementType" etc.

- ForProductionNumber: This name does not seem to indicate the meaning intended by VDV 454.

Slides 27/28:

- SIRI.Components has the meaning VDV.FoFahrzeugPositionen (slide 27) and VDV.FoFahrtAbschnittFahrzeugGruppen (slide 28).

- Components is very general in terms of terms. However, the specific intended use aims at the section of the journey. This is not reflected in the name.

Slide 29:
- CompoundTrain: "CoupledJourney" is used in NeTEx.

Slide 32:
- FormationCondition should not be a child element of JourneyPartInfo.

Additive:
In Sydney there are trains that are longer than some of the platforms, so that sometimes passengers cannot get on or off at the beginning or sometimes at the end of the train. However, this situation has not yet been provided for in VDV 454.

**Comment by Ulf9** [comment]
Sorry for using this channel in this way, but haeckerbaer: Can you please send me an email so that I get your current enail-address? The latest email-address to you bounces for some reason.

File: xsd/siri_utility/siri_types-v2.0.xsd (hunk 1 above)
Comment (haeckerbaer) [reply xsd/siri_utility/siri_types-v2.0.xsd:156]: @hennoit and @Aurige 
- ArrivalBoardingActivity annot. was updated
- annot. of both XOrientationRelativeToQuay were updated (more detailed examples)
- PercentageType was updated according to discussion on basecamp

**Comment by hennoit** [comment]
There is still one "arriving" which should be "departing" on line 557 of siri_journey-v2.0.xsd (DepartureOrientationRelativeToQuay annotation)

**Comment by EU-Agency-for-Railways** [comment]
I have reviewed the pull request from 
- in TrainElement/VehicleTypeGroup/VehiclePropertiesGroup the element EuroClass should be explained in more detail or renamed. I assume this is the European Identification Number (EIN) of the vehicle type (see https://www.era.europa.eu/sites/default/files/registers/docs/technical_doc_ein_codification_en.pdf)?
- in TrainElement / TrainElementGroup / ForProductionNumber has the wrong type TrainElementTypeEnumeration. Should be string with max 12 letters if the EVN number is mentioned here
-

**Comment by haeckerbaer** [comment]
@mentzrh 

- It was decided by the CEN group (see [discussion ](https://3.basecamp.com/3256016/buckets/9672657/messages/2804824485#__recording_2808201837)) that the names are not to be changed because they align with Transmodel/NeTEx (e.g. TrainElement instead of the proposed VehicleElement).
- ForProductionNumber is changed to VehicleNumber which corresponds to the EVN:
image
- Train.Components was changed to Train.TrainComponents and CompoundTrain.Components was changed to CompoundTrain.TrainsInCompoundTrain to better reflect the type of elements that are specified in those lists.
- CoupledJourney is an entirely different concept (through coaches). The name CompoundTrain is kept (see first point).
- FormationCondition was removed from the JourneyPartInfoStructure

**Comment by haeckerbaer** [comment]
@Ulf9, @hennoit and @EU-Agency-for-Railways 
Annotations were updated. 
The TrainElementCode is meant for the identifiert, e.g. "**ch:1:TrainElement:fz-f749fcec-2e65-426c-a8e1-45abc89d816c**". The ForProductionNumber was changed to VehicleNumber which corresponds to the UIC wagon number or EVN. ForAdvertisementNumber was removed. The annotation of EuroClass was updated.

**Comment by haeckerbaer** [comment]
I also changed the FormationAssignmentStructure and VehicleOccupancyStructure slightly. Instead of multiple QuayRefs one can now specify multiple StopAssignments within a FormationAssignment.

**Pull request merged**